vm/actor/src/builtin/power/state.rs

```diff
@@ -2,182 +2,100 @@
-    pub(super) fn add_miner_balance<BS: BlockStore>(
-        &mut self,
-        store: &BS,
-        miner: &Address,
-        amount: &TokenAmount,
-    ) -> Result<(), String> {
-        let mut bt = BalanceTable::from_root(store, &self.escrow_table)?;
-        bt.add(miner, amount)?;
-        self.escrow_table = bt.root()?;
-        Ok(())
-    }
-
-    /// Subtracts amount to miner balance at address using escrow table
-    #[allow(dead_code)]
-    pub(super) fn subtract_miner_balance<BS: BlockStore>(
-        &mut self,
-        store: &BS,
-        miner: &Address,
-        amount: &TokenAmount,
-        balance_floor: &TokenAmount,
-    ) -> Result<TokenAmount, String> {
-        let mut bt = BalanceTable::from_root(store, &self.escrow_table)?;
-        let amount = bt.subtract_with_minimum(miner, amount, balance_floor)?;
-        self.escrow_table = bt.root()?;
-        Ok(amount)
-    }
-    pub fn subtract_from_claim<BS: BlockStore>(
-        &mut self,
-        store: &BS,
-        miner: &Address,
-        power: &StoragePower,
-        pledge: &TokenAmount,
-    ) -> Result<(), String> {
-        let mut claim = self
-            .get_claim(store, miner)?
-            .ok_or(format!("no claim for actor {}", miner))?;
-
-        let old_nominal_power = self.compute_nominal_power(store, miner, &claim.power)?;
-
-        claim.power -= power;
-        claim.pledge = claim
-            .pledge
-            .checked_sub(pledge)
-            .ok_or("negative claimed pledge")?;
-
-        self.update_claim(store, miner, power, claim, old_nominal_power)
-    }
     pub fn add_to_claim<BS: BlockStore>(
         &mut self,
         store: &BS,
         miner: &Address,
         power: &StoragePower,
-        pledge: &TokenAmount,
+        qa_power: &StoragePower,
     ) -> Result<(), String> {
         let mut claim = self
             .get_claim(store, miner)?
             .ok_or(format!("no claim for actor {}", miner))?;
 
-        let old_nominal_power = self.compute_nominal_power(store, miner, &claim.power)?;
+        // update power
+        claim.raw_byte_power += power;
+        claim.quality_adj_power += qa_power;
 
-        claim.power += power;
-        claim.pledge += pledge;
-
-        self.update_claim(store, miner, power, claim, old_nominal_power)
-    }
-    /// Function will update the claim after `add_to_claim` or `subtract_from_claim` are called
-    /// * Logic is modified from spec to not use negative values
-    /// TODO revisit: logic for parts of this function seem wrong/ unnecessary
-    fn update_claim<BS: BlockStore>(
-        &mut self,
-        store: &BS,
-        miner: &Address,
-        power: &StoragePower,
-        claim: Claim,
-        old_nominal_power: StoragePower,
-    ) -> Result<(), String> {
-        let new_nominal_power = self.compute_nominal_power(store, miner, &claim.power)?;
+        let old_nominal_power = &claim.quality_adj_power;
```

Comment:
this needs to be before the changes in claim.quality_adj_power, otherwise logic is broken